### ISSUE TITLE
 Option to choose the colors of the arrows between nodes

### ISSUE THREAD

**Issue by Barbarezza** [YarnSpinnerTool/YarnEditor#270]
It could be very useful to quickly identify sections and have more visual clarity

**Comment by blurymind** [comment]
can you elaborate on that? Different arrow colour depending on node it comes from?

**Comment by Barbarezza** [comment]
More like the possibility of being able to independently choose the color of an arrow, a bit like it happens for the upper part of the nodes

**Comment by blurymind** [comment]
how would that work? it will use the same color as the node?

**Comment by FaultyFunctions** [comment]
Oh it would be cool if the arrow took the color of the starting node and faded into the color of the ending node?

**Comment by blurymind** [comment]
that would be awesome, but maybe should be optional? It may get a bit distracting with big files having rainbows all over the place :)

**Comment by FaultyFunctions** [comment]
Not sure why it would be done with CSS. This can be accomplished by adding a gradient to the html5 canvas when the arrows are being drawn inside `workspace.js`.

**Comment by blurymind** [comment]
Either way it would be worth investigating performance with the stress test
as well as somehow making it optional even somehow theme sensitive

**Comment by Barbarezza** [comment]
I apologize for the late reply, evidently I did not notice the email that advised me.

It would be very useful to be able to actually change the color of the arrow independently of the nodes, to be able to visualize perhaps some groupings as a whole or to be able to give meanings to colors (for example, a green arrow means access towards a good ending and a red arrow towards a bad ending, and having some independent colors for the nodes could give more visual information on the content of that node, for example "aggressive npc reaction" or "accommodating npc reaction")

**Comment by blurymind** [comment]
it could be that only selected nodes display that gradient indication too. Non selected ones render normally

**Comment by Barbarezza** [comment]
I totally agree that it should be optional in any case. Surely there would be those who would not use a similar feature and would indeed find it annoying

**Comment by FaultyFunctions** [comment]
Good ideas. Do you have any thoughts on how color selection for arrows would work? Currently arrows are drawn by looking at the "linked nodes" array list on individual nodes, so there is no actual data for the arrows themselves. Once the UI is figured out, that'll determine the implementation details I think.

**Comment by blurymind** [comment]
The nodes themselves already have a color selector.I think the node color should be what drives the gradient

**Comment by FaultyFunctions** [comment]
No I think @Barbarezza is suggesting you be able to choose the arrow color individually for more of an organizational benefit and not have it be linked to the nodes themselves.

**Comment by blurymind** [comment]
so by arrow color - its one color - not a gradient, correct? In that case we will need to be selective as to what colors can be set - similar to nodes - or we can straight out give you a color selector

**Comment by blurymind** [comment]
I am thinking of implementing this in a simpler way for now.

This is so the way the arrow is colored depends on the color of the node it targets.
A nice usability improvement would also be to have a tooltip of the node name when you hover over the arrow - maybe something to look into.
